apps/portal/src/plugins/data.tsx

```
@@ -30,62 +42,68 @@ export function usePluginDataRefresh(): () => void {
     source: "sample",
-    loading: githubInstalled,
+    loading: true,
   });
 
   useEffect(() => {
-    if (!githubInstalled) {
-      setState({ tasks: SAMPLE_TASKS, source: "sample", loading: false });
-      return;
-    }
     let alive = true;
     setState((s) => ({ ...s, loading: true }));
```

ESLint

[error] Error: Calling setState synchronously within an effect can trigger cascading renders

Effects are intended to synchronize state between React and external systems such as manually updating the DOM, state management libraries, or other platform APIs. In general, the body of an effect should do one or both of the following:
* Update external systems with the latest state from React.
* Subscribe for updates from some external system, calling setState in a callback function when external state changes.

Calling setState synchronously within an effect body causes cascading renders that can hurt performance, and is not recommended. (https://react.dev/learn/you-might-not-need-an-effect).

/home/jailuser/git/apps/portal/src/plugins/data.tsx:54:5
  52 |   useEffect(() => {
  53 |     let alive = true;
> 54 |     setState((s) => ({ ...s, loading: true }));
     |     ^^^^^^^^ Avoid calling setState() directly within an effect
  55 |     // Always ask the server: it aggregates owned tasks (when signed in) with any
  56 |     // connected source (GitHub). Only when nothing contributed do we show sample.
  57 |     getTasks()

(react-hooks/set-state-in-effect)